modules/renter/contractor/host_integration_test.go

```diff
@@ -472,3 +472,54 @@ func TestIntegrationModify(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// TestIntegrationRenew tests that the contractor can renew a previously-
+// formed file contract.
+func TestIntegrationRenew(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	// create testing trio
+	h, c, _, err := newTestingTrio("TestIntegrationRenew")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// get the host's entry from the db
+	hostEntry, ok := c.hdb.Host(h.ExternalSettings().NetAddress)
+	if !ok {
+		t.Fatal("no entry for host in db")
+	}
+
+	// form a contract with the host
+	contract, err := c.managedNewContract(hostEntry, modules.SectorSize*10, c.blockHeight+100)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// revise the contract
+	editor, err := c.Editor(contract)
+	if err != nil {
+		t.Fatal(err)
+	}
+	data, err := crypto.RandBytes(int(modules.SectorSize))
+	if err != nil {
+		t.Fatal(err)
+	}
+	// insert the sector
+	_, err = editor.Upload(data)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = editor.Close()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// renew the contract
+	contract = c.contracts[contract.ID]
+	_, err = c.managedRenew(contract, modules.SectorSize*10, c.blockHeight+200)
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
I would like to see this test extended, so that you do some checking on the values of the renewed contract. Might be good to do a download using the renewed contract as well, just to make sure that everything is at least working.